cpp/src/join/hash_join.cuh

```diff
@@ -62,11 +62,11 @@ namespace detail {
  * @return An estimate of the size of the output of the join operation
  */
 template <join_kind JoinKind, typename multimap_type>
-size_type estimate_join_output_size(table_device_view build_table,
-                                    table_device_view probe_table,
-                                    multimap_type const& hash_table,
-                                    null_equality compare_nulls,
-                                    rmm::cuda_stream_view stream)
+size_t estimate_join_output_size(table_device_view build_table,
```

Comment:
If you add the `std::` namespace to `size_t` then please use it consistently.

Reply:
Typically, we just use `size_t`. Whether `std::size_t` or `size_t` it is just personal preference.

Reply:
Fixed.

It seems like there's two conflicting views on `size_t` v/s `std::size_t` (see @jrhemstad's [comment](https://github.com/rapidsai/cudf/pull/8139#discussion_r625168022) above), where this might be good to clarify in the dev guide.